pkg/chunk/storage/caching_storage_client.go

```diff
@@ -0,0 +1,113 @@
+package storage
+
+import (
+	"bytes"
+	"context"
+	"strings"
+	"time"
+
+	"github.com/weaveworks/cortex/pkg/chunk"
+)
+
+type cachingStorageClient struct {
+	chunk.StorageClient
+	cache    *chunk.FifoCache
+	validity time.Duration
+}
+
+func newCachingStorageClient(client chunk.StorageClient, size int, validity time.Duration) chunk.StorageClient {
+	if size == 0 {
+		return client
+	}
+
+	return &cachingStorageClient{
+		StorageClient: client,
+		cache:         chunk.NewFifoCache("index", size, validity),
+		validity:      validity,
```

Comment:
Is validity needed on the `cachingStorageClient`? Or only to create the cache?

Reply:
You're right. It was done outside the cache before, now I've pushed it into the cache. Forgot to remove this.